source/common/network/address_impl.cc

```diff
@@ -261,6 +261,11 @@ PipeInstance::PipeInstance(const sockaddr_un* address, socklen_t ss_len)
 }
 
 PipeInstance::PipeInstance(const std::string& pipe_path) : InstanceBase(Type::Pipe) {
+  if (pipe_path.size() >= sizeof(address_.sun_path)) {
+    throw EnvoyException(
```

Comment:
Can we get an EXPECT_THROW_WITH_MESSAGE test for this?

Reply:
Done. I used `WITH_REGEX` to avoid making too many assumptions about `sizeof(sun_path)`.